zebra-consensus/src/block/subsidy/general.rs

```diff
@@ -102,6 +102,24 @@ pub fn find_output_with_amount(
         .collect()
 }
 
+/// Validate shielded consensus rules as described in [ZIP-213][ZIP-213]
+///
+/// [ZIP-213]: https://zips.z.cash/zip-0213#specification
+pub fn shielded_coinbase(height: Height, network: Network, transaction: &Transaction) -> bool {
+    let heartwood_height = Heartwood
+        .activation_height(network)
+        .expect("heartwood activation height should be available");
+
+    if height < heartwood_height {
+        if !transaction.joinsplits() && transaction.shields().is_none() {
```

Comment:
We can answer this question without a `shields()` method, by destructuring the struct to get `ShieldedData`. But the rule is not just 'no shielded data`, which is our construct, but no `Spend` descriptions, which we can get out `ShieldedData` via `spends()` which already exists. Example in https://github.com/ZcashFoundation/zebra/pull/1174/files#diff-f1a5abeca51e4ab069ec83ec7f1cdc6e10501ef76cffc7ddadf5dfa3e8a9a579R90

Reply:
I don't think we should be destructuring inside subsidy. If we do it to get `shielded_data` we will have to do it for `outputs()` or `joinsplits()` for consistency. I might be missing something but i see the method as the best option here.

I did some cleaning to the method and renamed to `shielded_data()` in https://github.com/ZcashFoundation/zebra/pull/1190/commits/a4d2644d7b052fd75c19b67ffd7b9c5e4f6809e8